models/turbine_models/custom_models/stateless_llama.py

```diff
@@ -489,26 +491,362 @@ def evict_kvcache_space(self):
+                total_time = time.time() - dec_time
+                yield self.tokenizer.decode(history), total_time
+
+            self.prev_token_len = token_len + len(history)
+
+            if format_out(token) == llm_model_map[self.hf_model_name]["stop_token"]:
+                break
+
+        for i in range(len(history)):
+            if type(history[i]) != int:
+                history[i] = int(history[i])
+        result_output = self.tokenizer.decode(history)
+        self.global_iter += 1
+        return result_output, total_time
+
 if __name__ == "__main__":
-    args = parser.parse_args()
-    mod_str, _ = export_transformer_model(
+    from turbine_models.custom_models.llm_cmd_opts import args
+    
+    mlir = args.input_mlir
+    vmfb = None
+    weight = None
+
+    flags = []
+    if "cpu" in args.device:
+        flags.extend(
+            [
+                "--iree-global-opt-enable-quantized-matmul-reassociation",
+            ]
+        )
+    elif args.device == "vulkan":
+        flags.extend(["--iree-stream-resource-max-allocation-size=4294967296"])
+    elif args.device == "rocm":
+        flags.extend(
+            [
+                "--iree-codegen-llvmgpu-enable-transform-dialect-jit=false",
+                "--iree-llvmgpu-enable-prefetch=true",
+                "--iree-opt-outer-dim-concat=true",
+                "--iree-flow-enable-aggressive-fusion",
+            ]
+        )
+        if "gfx9" in args.iree_target_triple:
+            flags.extend(
+                [
+                    f"--iree-codegen-transform-dialect-library={get_mfma_spec_path(args.iree_target_triple, get_checkpoints_path())}",
+                    "--iree-codegen-llvmgpu-use-vector-distribution=true",
+                ]
+            )
+    flags.extend(llm_model_map[args.hf_model_name]["compile_flags"])
+
+    if not args.pipeline_dir:
+        args.pipeline_dir = "./shark_vmfbs"
+    if not args.external_weights_dir and args.external_weights:
+        args.external_weights_dir = args.pipeline_dir
+
+    sd_pipe = StatelessLlamaPipeline(
         args.hf_model_name,
-        args.hf_auth_token,
-        args.compile_to,
-        args.external_weights,
-        args.external_weight_file,
-        args.quantization,
+        args.scheduler_id,
+        args.height,
+        args.width,
         args.precision,
+        args.max_length,
+        args.batch_size,
+        args.num_inference_steps,
         args.device,
         args.iree_target_triple,
-        args.vulkan_max_allocation,
-        args.streaming_llm,
-        args.vmfb_path,
-        upload_ir=False,
-        decomp_attn=args.decomp_attn,
+        flags,
+        args.attn_spec,
+        args.decomp_attn,
+        args.pipeline_dir,
+        args.external_weights_dir,
+        args.external_weights,
+        args.vae_decomp_attn,
+        args.hf_auth_token,
+    )
+    vmfb, weight = sd_pipe.check_prepared(mlir, vmfb, weight, interactive=False, quantization="int4")
```

Comment:
this should be a flag

Reply:
might make more sense to rename it to "prepare_pipeline" since this checks for vmfbs, and exports them if they aren't found.